### DISCUSSION TITLE
help for new comers

### DISCUSSION THREAD

**Discussion by smagtech** [pschatzmann/ESP32-A2DP#502] [Q&A] [answered]
I'm getting this error can anybody help please. 

Arduino
Version: 2.2.1
CLI Version: 0.34.0

error is....

first red flag....

In file included from c:\Users\G_Sudhir\Documents\Arduino\libraries\arduino-audio-tools-main\src/AudioLibs/AudioKit.h:4,
                 from C:\Users\G_Sudhir\AppData\Local\Temp\.arduinoIDE-unsaved20231012-3424-11tccjn.j7oj\basic-a2dp-audiokit\basic-a2dp-audiokit.ino:10:
c:\Users\G_Sudhir\Documents\Arduino\libraries\arduino-audiokit-main\src/AudioKitHAL.h: In member function 'i2s_config_t audiokit::AudioKitConfig::i2sConfig()':
c:\Users\G_Sudhir\Documents\Arduino\libraries\arduino-audiokit-main\src/AudioKitHAL.h:123:9: warning: missing initializer for member 'i2s_driver_config_t::fixed_mclk' [-Wmissing-field-initializers]
         };
         ^
c:\Users\G_Sudhir\Documents\Arduino\libraries\arduino-audiokit-main\src/AudioKitHAL.h:123:9: warning: missing initializer for member 'i2s_driver_config_t::mclk_multiple' [-Wmissing-field-initializers]
c:\Users\G_Sudhir\Documents\Arduino\libraries\arduino-audiokit-main\src/AudioKitHAL.h:123:9: warning: missing initializer for member 'i2s_driver_config_t::bits_per_chan' [-Wmissing-field-initializers]

second red flag.... 

c:\Users\G_Sudhir\Documents\Arduino\libraries\ESP32-A2DP-main\src\BluetoothA2DPSinkQueued.cpp: In member function 'virtual void BluetoothA2DPSinkQueued::i2s_task_handler(void*)':
c:\Users\G_Sudhir\Documents\Arduino\libraries\ESP32-A2DP-main\src\BluetoothA2DPSinkQueued.cpp:52:12: error: unused variable 'bytes_written' [-Werror=unused-variable]
     size_t bytes_written = 0;
            ^~~~~~~~~~~~~
cc1plus.exe: some warnings being treated as errors
c:\Users\G_Sudhir\Documents\Arduino\libraries\ESP32-A2DP-main\src\BluetoothA2DPSink.cpp: In member function 'virtual void BluetoothA2DPSink::audio_data_callback(const uint8_t*, uint32_t)':
c:\Users\G_Sudhir\Documents\Arduino\libraries\ESP32-A2DP-main\src\BluetoothA2DPSink.cpp:1150:10: error: variable 'is_callback_used' set but not used [-Werror=unused-but-set-variable]
     bool is_callback_used = false;
          ^~~~~~~~~~~~~~~~
cc1plus.exe: some warnings being treated as errors

exit status 1

Compilation error: exit status 1

**Comment by pschatzmann** [comment, marked as answer]
I guess you changed the Preferences in Arduino and you treat warnings as errors. 
Just change it back to Default!

By the way you forgot to provide the most relevant information: what ESP32 Core version do you use ?
